lib/spack/spack/build_systems/cuda.py

```diff
@@ -0,0 +1,71 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os
+from spack import *
+
+
+class CudaPackage(Package):
+    """Auxiliary class which contains CUDA variant, dependencies and conflicts
+    and is meant to unify and facilitate its usage.
+    """
+
+    # FIXME: keep cuda and cuda_arch separate to make usage easier untill
+    # Spack has depends_on(cuda, when='cuda_arch!=None') or alike
+    variant('cuda', default=False,
+            description='Build with CUDA')
+    # see http://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#gpu-feature-list
+    variant('cuda_arch', default=None,
+            description='CUDA architecture',
+            values=('20', '30', '32', '35', '50', '52', '53', '60', '61', '62', '70'),
```

Comment:
what would be nice to have for derived packages is
```
conflicts('+cuda', when='cuda_arch<35')
```
or alike to express that we need `35` or higher. This is somewhat related to https://github.com/LLNL/spack/issues/6050

Reply:
Doesn't the `when` alternatively allow for a boolean value?  If so, *and* there is a way to get `cuda_arch` directly (no `self.spec.cuda_arch` yet right?), it could just be `conflicts('+cuda', when=(magic.cuda_arch < "35"))`.  ASCII comparison should be OK

Reply:
unfortunately i don't think it is possible.

@tgamblin @alalazo that probably requires `conflicts` and `depends_on` to have access to `spec` in `when` part.